test/tools/test_all_collections.py

```diff
@@ -21,8 +21,7 @@ def test(yamlfile_path, cxxfile_path):
     datatypes_found = []
 
     with open(cxxfile_path, mode='r', encoding="utf-8") as cxxfile:
-        for cxxline in cxxfile:
-            cxxline = cxxfile.readline()
+        for cxxline in cxxfile.readlines():
```

Comment:
This prevents accidentally skipping every other line(!), which worked before because we were and now we removed an odd number of lines, so it broke for this PR.